pandas/core/style.py

```diff
@@ -255,6 +272,75 @@ def _translate(self):
                     precision=precision, table_styles=table_styles,
                     caption=caption, table_attributes=self.table_attributes)
 
+    def format(self, formatter, subset=None):
+        '''
+        Format the text display value of cells.
+
+        .. versionadded:: 0.18.0
+
+        Parameters
+        ----------
+        formatter: str, callable, or dict
+        subset: IndexSlice
+
+        Returns
+        -------
+        self: Styler
+
+        Notes
+        -----
+
+        ``formatter`` is either an ``a`` or a dict ``{column name: a}`` where
+        ``a`` is one of
+            - str: this will be wrapped in: ``a.format(x)``
+            - callable: called with the value of an individual cell
+
+        The default display value for numeric values is the "general" (``g``)
+        format with ``pd.options.display.precisoin`` precisoin.
+
+        Examples
+        --------
+
+        >>> df = pd.DataFrame(np.random.randn(4, 2), columns=['a', 'b'])
+        >>> df.style.format("{:.2%}")
+        >>> df['c'] = ['a', 'b', 'c', 'd']
+        >>> df.style.format({'C': str.upper})
+        '''
+        if subset is not None and not com.is_sequence(subset):
+            subset = [subset]
+
+        just_row_subset = (
+            (subset is not None and not com.is_sequence(subset)) or
+            (com.is_list_like(subset) and not isinstance(subset, tuple))
+        )
+
+        if just_row_subset:
+            row_locs = self.data.index.get_indexer_for(subset)
+            col_locs = np.arange(len(self.data.columns))
```

Comment:
prob should have this logic generally available for subsetting ops (though you did somewhere?)